Kudu.Contracts/Settings/DeploymentSettingsExtension.cs

```diff
@@ -75,7 +76,26 @@ public static TimeSpan GetLogStreamTimeout(this IDeploymentSettingsManager setti
 
         public static TimeSpan GetHttpClientTimeout(this IDeploymentSettingsManager settings)
         {
-            return GetTimeSpan(settings, SettingsKeys.HttpClientTimeout, DefaultHttpClientTimeout);
+            TimeSpan defaultTimeout = DefaultHttpClientTimeout;
+
+            if (IsMoonCake())
+            {
+                // Update default timeout for SettingsKeys.HttpClientTimeout for Mooncake
+                defaultTimeout = MoonCakeDefaultHttpClientTimeout;
+            }
+
+            return GetTimeSpan(settings, SettingsKeys.HttpClientTimeout, defaultTimeout);
+        }
+
+        private static bool IsMoonCake()
```

Comment:
As a habit, make this a public and add unit test.   It served as documentation and what to expect.  I would make an API to take string  eg. IsMoonCake(string currentStampName).   This allows unit testing w/o munging with env.

Reply:
Changed the method to be test friendly. also added a Fact test for that method.